codegen/thrift.go

```diff
@@ -133,3 +133,93 @@ func isHashable(spec compile.TypeSpec) bool {
 		return false
 	}
 }
+
+func pointerMethodType(typeSpec compile.TypeSpec) string {
+	var pointerMethod string
+
+	switch typeSpec.(type) {
+	case *compile.BoolSpec:
+		pointerMethod = "Bool"
+	case *compile.I8Spec:
+		pointerMethod = "Int8"
+	case *compile.I16Spec:
+		pointerMethod = "Int16"
+	case *compile.I32Spec:
+		pointerMethod = "Int32"
+	case *compile.I64Spec:
+		pointerMethod = "Int64"
+	case *compile.DoubleSpec:
+		pointerMethod = "Float64"
+	case *compile.StringSpec:
+		pointerMethod = "String"
+	default:
+		panic(fmt.Sprintf(
+			"Unknown type (%T) %v for allocating a pointer",
+			typeSpec, typeSpec,
+		))
+	}
+
+	return pointerMethod
+}
+
+type walkFieldVisitor func(
+	goPrefix string,
+	thriftPrefix string,
+	field *compile.FieldSpec,
+) bool
+
+func walkFieldGroups(
+	fields compile.FieldGroup,
+	visitField walkFieldVisitor,
+) bool {
+	return walkFieldGroupsInternal("", "", fields, visitField)
+}
+
+func walkFieldGroupsInternal(
+	goPrefix string,
+	thriftPrefix string,
+	fields compile.FieldGroup,
+	visitField walkFieldVisitor,
+) bool {
+	for i := 0; i < len(fields); i++ {
+		field := fields[i]
+
+		bail := visitField(goPrefix, thriftPrefix, field)
+		if bail {
+			return true
+		}
+
+		realType := compile.RootTypeSpec(field.Type)
+		switch t := realType.(type) {
+		case *compile.BinarySpec:
+		case *compile.StringSpec:
+		case *compile.BoolSpec:
+		case *compile.DoubleSpec:
+		case *compile.I8Spec:
+		case *compile.I16Spec:
+		case *compile.I32Spec:
+		case *compile.I64Spec:
+		case *compile.EnumSpec:
+		case *compile.StructSpec:
+			bail := walkFieldGroupsInternal(
+				goPrefix+"."+pascalCase(field.Name),
+				thriftPrefix+"."+field.Name,
+				t.Fields,
+				visitField,
+			)
+			if bail {
+				return true
+			}
+		case *compile.SetSpec:
+			// TODO: implement
```

Comment:
Should we panic on unimplemented types so that they are rejects in the UI continuous integration?